src/couch_replicator/src/couch_replicator_httpc.erl

```diff
@@ -139,12 +139,12 @@ process_response({error, sel_conn_closed}, Worker, HttpDb, Params, _Cb) ->
 %% next request.
 process_response({error, connection_closing}, Worker, HttpDb, Params, _Cb) ->
     stop_and_release_worker(HttpDb#httpdb.httpc_pool, Worker),
-    throw({retry, HttpDb, Params});
+    maybe_retry({error, connection_closing}, Worker, HttpDb, Params);
 
-process_response({error, req_timedout}, _Worker, HttpDb, Params, _Cb) ->
+process_response({error, req_timedout}, Worker, HttpDb, Params, _Cb) ->
     % ibrowse worker terminated because remote peer closed the socket
     % -> not an error
-    throw({retry, HttpDb, Params});
+    maybe_retry({error, req_timedout}, Worker, HttpDb, Params);
```

Comment:
Shouldn't we just remove this clause entirely and let it fall through to the catch-all at the bottom?

Reply:
Oh totally, let's remove it. Good call.